common/kv_db/in_memory/in_memory.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <common/kv_db.hpp>
+#include <common/kv_db_in_memory.hpp>
+
+#include <string>
+
+namespace mender::common::kv_db {
+
+using namespace std;
+
+class InMemoryTransaction : public Transaction {
+public:
+	InMemoryTransaction(KeyValueDataBaseInMemory &db, bool read_only);
+
+	ExpectedBytes Read(const string &key) override;
+	ExpectedBool Write(const string &key, const vector<byte> &value) override;
+	ExpectedBool Remove(const string &key) override;
+
+private:
+	KeyValueDataBaseInMemory &db_;
+	bool read_only_;
+};
+
+InMemoryTransaction::InMemoryTransaction(KeyValueDataBaseInMemory &db, bool read_only) :
+	db_(db),
+	read_only_(read_only) {
+}
```

Comment:
Could be part of the class declaration above.

Reply:
Good point, I keep forgetting that this is allowed now.